ParcelDistributionCenter/ParcelDistributionCenter.Logic/Services/CourierService.cs

```diff
@@ -1,39 +1,84 @@
 using ParcelDistributionCenter.Model.Repositories;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
 
 namespace ParcelDistributionCenter.Logic.Services
 {
     // TODO: Prevent code from nullable ids coming form json
     public class CourierService : ICourierService
     {
-        private readonly IPackageServices _packageServices;
-        private readonly IRepository<Courier> _repository;
+        private readonly IRepository<Courier> _courierRepository;
+        private readonly IRepository<Package> _packageRepository;
 
-        public CourierService(IRepository<Courier> repository, IPackageServices packageServices)
+        public CourierService(IRepository<Courier> courierRepository, IRepository<Package> packageRepository)
         {
-            _repository = repository;
-            _packageServices = packageServices;
+            _courierRepository = courierRepository;
+            _packageRepository = packageRepository;
         }
 
-        public void Delete(Courier model)
+        public bool DeleteCourier(string id)
         {
-            var courier = FindById(model.CourierJsonId);
-            _packageServices.UnassignCouriersPackages(courier.CourierJsonId);
-            _repository.Delete(courier);
+            var courier = FindById(id);
+            if (courier != null)
+            {
+                UnassignCouriersPackages(id);
+                _courierRepository.Delete(courier);
+                return true;
+            }
+            return false;
         }
 
-        public IEnumerable<Courier> FindAll() => _repository.GetAll();
+        public IEnumerable<Courier> GetAll() => _courierRepository.GetAll();
 
-        public Courier FindById(string id) => _repository.Get(id);
+        public List<Package> GetCourierPackages(string courierId)
+        {
+            var packages = _packageRepository.GetAll().Where(p => p.CourierId == courierId).ToList();
+
+            return packages;
+        }
+
+        public Courier FindById(string id) => _courierRepository.Get(id);
+
+        public void UnassignCouriersPackages(string CourierId)
+        {
+            IEnumerable<Package> packages = _packageRepository.GetAll().Where(p => p.CourierId == CourierId);
+            packages.Select(p => p.CourierId = "Unassigned");
+            foreach(Package p in packages)
+            {
+                _packageRepository.Update(p);
+            }
+        }
 
         public void Update(Courier model)
         {
-            var courier = FindById(model.CourierJsonId);
+            var courier = FindById(model.Id);
             courier.Name = model.Name;
             courier.Surname = model.Surname;
             courier.Email = model.Email;
             courier.Phone = model.Phone;
+            _courierRepository.Update(courier);
+
+        }
+
+        public IEnumerable<Package> GetUnassignedPackages()
+        {
+            return _packageRepository.GetAll().Where(p => p.CourierId == null);
```

Comment:
Jak nie za długie, to może też wyrażenie lambda?

Reply:
Poprawione